File: Elementary Cellular Automaton/simulation.js
const DIMENSION = 800;
const RESOLUTION = 50; // whole number
const CELL_WIDTH = DIMENSION / RESOLUTION;

let board = [];
let current_line = 1;

function setup() {
  createCanvas(DIMENSION, DIMENSION / 2);
  for (let i = 0; i < RESOLUTION; i++) {
    board.push([]);
    for (let j = 0; j < RESOLUTION; j++) {
      board[i].push(0);
    }
  }
  // Initialize top cell
  board[RESOLUTION / 2][0] = 1;
}

function draw() {
  background(255);
  for (let i = 0; i < RESOLUTION; i++) {
    for (let j = 0; j < RESOLUTION; j++) {
      if (board[i][j] == 1) {
        fill(0);
        square(i * CELL_WIDTH, j * CELL_WIDTH, CELL_WIDTH);
      }
    }
  }
  new_board(current_line);
  current_line++;
}

function neighbours_color(i, j) {
  let neighbours = [];
  if (board[i - 1][j - 1] == 1) {
    neighbours.push(1);
  } else {
    neighbours.push(0);
  }
  if (board[i][j - 1] == 1) {
    neighbours.push(1);
  } else {
    neighbours.push(0);
  }
  if (board[i + 1][j - 1] == 1) {
    neighbours.push(1);
  } else {
    neighbours.push(0);
  }
  return neighbours;
}

function set_rule(i, j) {
  let neighbours = neighbours_color(i, j);
  
  // Rule 182
  if (are_equal(neighbours, [1, 1, 1])) {
    board[i][j] = 1;
  } else if (are_equal(neighbours, [1, 1, 0])) {
    board[i][j] = 0;
  } else if (are_equal(neighbours, [1, 0, 1])) {
    board[i][j] = 1;
  } else if (are_equal(neighbours, [1, 0, 0])) {
    board[i][j] = 1;
  } else if (are_equal(neighbours, [0, 1, 1])) {
    board[i][j] = 0;
  } else if (are_equal(neighbours, [0, 1, 0])) {
    board[i][j] = 1;
  } else if (are_equal(neighbours, [0, 0, 1])) {
    board[i][j] = 1;
  } else if (are_equal(neighbours, [0, 0, 0])) {
    board[i][j] = 0;
  }
}

function new_board(current_line) {
  for (let i = 1; i < RESOLUTION - 1; i++) {
    set_rule(i, current_line);
  }
}

function are_equal(a1, a2) {
  for (let i = 0; i < a1.length; i++) {
    if (a1[i] != a2[i]) {
      return false;
    }
  }
  return true;
}

File: Elementary Cellular Automaton/simulation.py
import pygame


def neighbours_color(i, j):
    neighbours = []
    if board[i - 1][j - 1] == 1:
        neighbours.append(1)
    else:
        neighbours.append(0)
    if board[i][j - 1] == 1:
        neighbours.append(1)
    else:
        neighbours.append(0)
    if board[i + 1][j - 1] == 1:
        neighbours.append(1)
    else:
        neighbours.append(0)
    return neighbours


def set_rule(i, j):
    neighbours = neighbours_color(i, j)

    if neighbours == [1, 1, 1]:
        board[i][j] = 0
    elif neighbours == [1, 1, 0]:
        board[i][j] = 1
    elif neighbours == [1, 0, 1]:
        board[i][j] = 0
    elif neighbours == [1, 0, 0]:
        board[i][j] = 0
    elif neighbours == [0, 1, 1]:
        board[i][j] = 1
    elif neighbours == [0, 1, 0]:
        board[i][j] = 1
    elif neighbours == [0, 0, 1]:
        board[i][j] = 1
    else:
        board[i][j] = 1


def new_board(current_line):
    for i in range(1, RESOLUTION - 1):
        set_rule(i, current_line)


WHITE = (255, 255, 255)
BLACK = (0, 0, 0)

DIMENSION_RATIO = 2
SCREEN_HEIGHT = 700
SCREEN_WIDTH = SCREEN_HEIGHT * DIMENSION_RATIO

RESOLUTION = 300  # on height
CELL_WIDTH = int(SCREEN_WIDTH / RESOLUTION)

launched = True
board = [[0 for _ in range(RESOLUTION * DIMENSION_RATIO)] for _ in range(RESOLUTION)]
board[int(RESOLUTION / 2)][0] = 1
current_line = 1

pygame.init()
screen = pygame.display.set_mode((SCREEN_WIDTH, SCREEN_HEIGHT))
pygame.display.set_caption("Title")

while launched:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            launched = False
    screen.fill(WHITE)
    for i in range(len(board)):
        for j in range(len(board[i])):
            if board[i][j] == 1:
                pygame.draw.rect(screen, BLACK, [i * CELL_WIDTH, j * CELL_WIDTH, CELL_WIDTH, CELL_WIDTH])
    new_board(current_line)
    current_line += 1
    pygame.display.update()
   
